src/codec_aom.c

```diff
@@ -737,12 +739,12 @@ static avifResult aomCodecEncodeImage(avifCodec * codec,
         if (encoder->maxThreads > 1) {
             aom_codec_control(&codec->internal->encoder, AV1E_SET_ROW_MT, 1);
         }
-        if (encoder->tileRowsLog2 != 0) {
-            int tileRowsLog2 = AVIF_CLAMP(encoder->tileRowsLog2, 0, 6);
+        if (tileRowsLog2 != 0) {
+            tileRowsLog2 = AVIF_CLAMP(tileRowsLog2, 0, 6);
             aom_codec_control(&codec->internal->encoder, AV1E_SET_TILE_ROWS, tileRowsLog2);
         }
-        if (encoder->tileColsLog2 != 0) {
-            int tileColsLog2 = AVIF_CLAMP(encoder->tileColsLog2, 0, 6);
+        if (tileColsLog2 != 0) {
+            tileColsLog2 = AVIF_CLAMP(tileColsLog2, 0, 6);
```

Comment:
nit/optional: seems like clamping is an AV1 spec thing. so we can move the clamping logic outside of the codec_* wrappers and just have it in one place.